src/dmd/attrib.d

```diff
@@ -1577,3 +1573,32 @@ int foreachUda(Dsymbol sym, Scope* sc, int delegate(Expression) dg)
         });
     });
 }
+
+/**
+ * Iterates the UDAs attached to the given symbol, without performing semantic
+ * analysis.
+ *
+ * Use this function instead of `foreachUda` if semantic analysis of `sym` is
+ * still in progress.
+ *
+ * Params:
+ *  sym = the symbol to get the UDAs from
+ *  dg = called once for each UDA
+ *
+ * Returns:
+ *  If `dg` returns `!= 0`, stops the iteration and returns that value.
+ *  Otherwise, returns 0.
+ */
+int foreachUdaNoSemantic(Dsymbol sym, int delegate(Expression) dg)
```

Comment:
Can't we have an `opApply`? Probably because of bootstrapping restrictions?

Reply:
There are two iteration functions (`foreachUda` and `foreachUdaNoSemantic`) and only one of them can be `UserAttributeDeclaration.opApply`. Most likely, it would be `foreachUda`, so this would need to be a free function regardless.

Reply:
Ah, makes sense